src/index.js

```diff
@@ -0,0 +1,130 @@
+          </div>
+          `
+  ) : (
+    `
+          <div class="sides__container">
+            <div class="toDo__left">
+              <i data-id="${task.id}" class="fa-regular fa-square completed" title="Mark as complete"></i>
+              <textarea name="textarea" class="toDo__description" cols="30" rows="1" readonly>${task.description}</textarea>
+            </div>
+            <div class="toDo__right">
+              <i data-id="${task.id}" class="fa-solid fa-trash-can remove" title="Delete Task"></i>
+              <i data-id="${task.id}" class="fa-solid fa-pen-to-square edit" title="Edit Task"></i>
+            </div> 
+          </div>
+          `)}
+      `;
+      tasksContainer.appendChild(li);
+    });
+  }
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  console.log(tasks);
+};
+
+const completeTask = (e) => {
+  if (e.target.classList.contains('completed')) {
+    const taskId = Number(e.target.getAttribute('data-id'));
+    tasks.forEach((it) => {
+      if (it.id === taskId) {
+        it.completed = !it.completed;
+      }
+    });
+  }
+  printTasks();
+};
+
+const removeTask = (el) => {
+  if (el.target.classList.contains('remove')) {
+    const taskId = Number(el.target.getAttribute('data-id'));
+    tasks.forEach((task, index) => {
+      if (task.id === taskId) {
+        tasks.splice(index, 1);
+        for (let i = index; i < tasks.length; i + 1) {
+          tasks[i].id -= 1;
+        }
+      }
+    });
+  }
+  printTasks();
+};
+
+const editTask = (el) => {
+  const description = document.querySelectorAll('.toDo__description');
+  if (el.target.classList.contains('edit')) {
+    const taskId = Number(el.target.getAttribute('data-id'));
+    tasks.forEach((task) => {
+      if (task.id === taskId) {
+        description[taskId - 1].removeAttribute('readonly');
+        description[taskId - 1].select();
+
+        description[taskId - 1].addEventListener('keydown', (k) => {
+          if (k.key === 'Enter') {
+            description[taskId - 1].setAttribute('readonly', 'readonly');
+            const newTasks = JSON.parse(localStorage.getItem('tasks'));
+            newTasks.forEach((task) => {
+              if (task.id === taskId) {
+                task.description = description[taskId - 1].value;
+              }
+            });
+            tasks = newTasks;
+            printTasks();
+          }
+        });
+      }
+    });
+  }
+};
+
+tasksContainer.addEventListener('click', completeTask);
+tasksContainer.addEventListener('click', removeTask);
+tasksContainer.addEventListener('click', editTask);
+toDoSubmit.addEventListener('click', (e) => {
+  e.preventDefault();
+  setTask();
+  printTasks();
+  addList.reset();
+});
+removeAll.addEventListener('click', () => {
+  tasks = [];
+  printTasks();
+});
+document.addEventListener('DOMContentLoaded', () => {
+  const data = JSON.parse(localStorage.getItem('tasks'));
+  tasks = data;
```

Comment:
- Good job done so far, but I also have errors when I try adding a task, see the image below, the reason for this is that When the DOM loads, you are reassigning the value of tasks to the value gotten from the local storage. But if the user viewing/testing your project does not have a task array in the local storage, it sets the value of tasks to null. To fix this issue, please remove the code on line 127 and 128, then on line 10:
` let tasks = JSON.parse(localStorage.getItem('tasks')) || []` 
![image](https://github.com/Feliperto/ToDo_List/assets/73607512/0a49812d-e88b-4f02-905a-4fe4d9b404f5)
- After doing the above, please make sure all other functionalities work as expected 🙏